internal/controllers/ipaddressclaim.go

```diff
@@ -256,3 +262,29 @@ func resourceTransitionedToUnpaused() predicate.Predicate {
 		},
 	}
 }
+
+func poolStatus(o client.Object) *v1alpha2.InClusterIPPoolStatusIPAddresses {
+	if ipPool, ok := o.(*v1alpha2.InClusterIPPool); ok {
```

Comment:
the linter is right, a type switch would be cleaner here.
https://gobyexample.com/switch (last example)

Reply:
Thanks for pointing it out. I've got it fixed.